### PR TITLE
Tweak to ATM websocket Disconnected 50x handling

### PR DESCRIPTION
**Pull request by SimonWoolf**
For compliance with https://github.com/ably/docs/pull/631

### CODE REVIEW

File: common/lib/transport/connectionmanager.js [hunk 1]
```diff
@@ -647,8 +648,24 @@ var ConnectionManager = (function() {
...
+			 * applies to connectBase unless the stored preference transport doesn't
+			 * work. We solve this by unpersisting the transport preference and
+			 * setting an instance variable to force fallback hosts to be used (if
+			 * any) here. Bit of a kludge, but no real better alternatives without
+			 * rewriting the entire thing */
+			if(state === 'disconnected' && error && error.statusCode > 500) {
+				this.unpersistTransportPreference();
+				this.forceFallbackHost = true;
+			}
 			/* TODO remove below line once realtime sends token errors as DISCONNECTEDs */
```
Comment (paddybyers) [inline common/lib/transport/connectionmanager.js]: Does this still need to be done?

File: spec/realtime/failure.test.js [hunk 2]
```diff
@@ -359,5 +360,33 @@ define(['ably', 'shared_helper', 'async'], function(Ably, helper, async) {
...
+				connection.once(function(stateChange) {
+					test.equal(stateChange.current, 'disconnected', 'expect next connection attempt to fail due to using the (bad) fallback host')
+					closeAndFinish(test, realtime);
+				});
+			});
+			connectionManager.activeProtocol.getTransport().onProtocolMessage(createPM({
+				action: 6,
+				error: {
+					message: "fake placement constraint",
+					code: 50330,
```
Comment (paddybyers) [inline spec/realtime/failure.test.js]: So this this work yet with 4xxxxx? Is there a plan to do that, or an issue?

File: common/lib/transport/connectionmanager.js (hunk 1 above)
Comment (SimonWoolf) [reply common/lib/transport/connectionmanager.js]: Yup, that's waiting on https://github.com/ably/realtime/issues/1154

Comment (SimonWoolf) [reply common/lib/transport/connectionmanager.js]: actually, wait, no, that's a different thing. Not sure, will check.

File: spec/realtime/failure.test.js (hunk 2 above)
Comment (SimonWoolf) [reply spec/realtime/failure.test.js]: as discussed, https://github.com/ably/realtime/issues/1154 will make it work with 4xxxx

Comment (mattheworiordan) [reply spec/realtime/failure.test.js]: Code should be 50320

**Pull request merged**